Sources/Machine/abstract_machine.hpp

```diff
@@ -230,12 +237,41 @@ class AbstractMachine {
 
   virtual ~AbstractMachine() = default;
 
+  void SetMaxBatchSize(Index max_batch_size) {
+    NETKET_CHECK(max_batch_size >= 1, InvalidInputError,
+                 "invalid maximum batch size: "
+                     << max_batch_size
+                     << "; expected a number larger than one");
+    max_batch_size_ = max_batch_size;
+  }
+
+  Index GetMaxBatchSize() const { return max_batch_size_; }
+
  protected:
   AbstractMachine(std::shared_ptr<const AbstractHilbert> hilbert)
-      : hilbert_(std::move(hilbert)) {}
+      : hilbert_(std::move(hilbert)), max_batch_size_(32) {}
+
+  // Function to dispatch sub batches call
+  // This is useful to make sure that batched calls on matrices are never called
+  // for more than max_batch_size_ rows at the time
+  template <class T1, class T2>
```

Comment:
I change this a bit now, let me know if you think of a way to remove completely the templates...

Reply:
Is there a reason why you'd like to get rid of templates here?

Reply:
I had the impression that this was slowing down compilation a bit, but maybe it's because the subbatching should be implemented in the AbstractMachine only and not in all the individual machines (now this is implemented separately for RbmSpin, but that's because this is WIP)